sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -639,7 +643,7 @@ def job_id(self):
     return self._job.id
 
   def metrics(self):
-    return DataflowMetrics()
+    return getattr(self, 'metric_results', None)
```

Comment:
why not just return self.metric_results? Why would it be undefined.

Or perhaps use `has_job` if this is a matter of having a job.